pynest/examples/Potjans_2014/network.py

```diff
@@ -19,486 +19,502 @@
-        to them. The initial membrane potential of the neurons is drawn from a
-        normal distribution. Scaling of the number of neurons and of the
-        synapses is performed. If scaling is performed extra DC input is added
-        to the neuronal populations.
-
+        to them. The initial membrane potential of the neurons is drawn from
+        normal distributions dependent on the parameter ``V0_type``.
         """
-        self.N_full = self.net_dict['N_full']
-        self.N_scaling = self.net_dict['N_scaling']
-        self.K_scaling = self.net_dict['K_scaling']
-        self.synapses = get_total_number_of_synapses(self.net_dict)
-        self.synapses_scaled = self.synapses * self.K_scaling
-        self.nr_neurons = self.N_full * self.N_scaling
-        self.K_ext = self.net_dict['K_ext'] * self.K_scaling
-        self.w_from_PSP = get_weight(self.net_dict['PSP_e'], self.net_dict)
-        self.weight_mat = get_weight(
-            self.net_dict['PSP_mean_matrix'], self.net_dict
-            )
-        self.weight_mat_std = self.net_dict['PSP_std_matrix']
-        self.w_ext = self.w_from_PSP
-        if self.net_dict['poisson_input']:
-            self.DC_amp_e = np.zeros(len(self.net_dict['populations']))
-        else:
-            if nest.Rank() == 0:
-                print(
-                    """
-                    no poisson input provided
-                    calculating dc input to compensate
-                    """
-                    )
-            self.DC_amp_e = compute_DC(self.net_dict, self.w_ext)
-
-        v0_type_options = ['original', 'optimized']
-        if self.net_dict['V0_type'] not in v0_type_options:
-            print(
-                '''
-                '{0}' is not a valid option, replacing it with '{1}'
-                Valid options are {2}
-                '''.format(self.net_dict['V0_type'],
-                           v0_type_options[0],
-                           v0_type_options)
-                )
-            self.net_dict['V0_type'] = v0_type_options[0]
         if nest.Rank() == 0:
-            print(
-                'The number of neurons is scaled by a factor of: %.2f'
-                % self.N_scaling
-                )
-            print(
-                'The number of synapses is scaled by a factor of: %.2f'
-                % self.K_scaling
-                )
-
-        # Scaling of the synapses.
-        if self.K_scaling != 1:
-            synapses_indegree = self.synapses / (
-                self.N_full.reshape(len(self.N_full), 1) * self.N_scaling)
-            self.weight_mat, self.w_ext, self.DC_amp_e = adj_w_ext_to_K(
-                synapses_indegree, self.K_scaling, self.weight_mat,
-                self.w_from_PSP, self.DC_amp_e, self.net_dict, self.stim_dict
-                )
-
-        # Create cortical populations.
+            print('Creating neuronal populations.')
+
         self.pops = []
         pop_file = open(
```

Comment:
Now I write all node ids to file on Rank 0 after all populations are created. The benefit is that only Rank 0 has to do it, but another loop is needed. Before, I did not want to use `with open()` because I did not want to indent the code block any further, but this solution seems to be better. What do you think?